### PR TITLE
close active listener when removing connection

### PR DESCRIPTION
**Pull request by wasaga**
## Summary

If there is an active listener associated with connection record,
previously it would be left active when connection is deleted, blocking the port 
until application is restarted. 

## Related issues
Fixes https://github.com/pomerium/internal/issues/716

## Checklist

- [x] reference any related issues
- [ ] updated docs
- [x] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: api/config_server.go [hunk 1]
```diff
@@ -41,8 +41,16 @@ func (s *server) Delete(_ context.Context, sel *pb.Selector) (*pb.DeleteRecordsR
 		return nil, err
 	}
 
+	ids := make([]string, 0, len(recs))
 	for _, r := range recs {
-		if err = s.config.delete(r.GetId()); err != nil {
+		ids = append(ids, r.GetId())
+	}
+	if _, err = s.disconnectLocked(ids); err != nil {
```
Comment (calebdoxsey) [inline api/config_server.go:48]: Does disconnect return an error if a connection isn't already connected?

https://github.com/pomerium/cli/blob/ae250e91e9b616e2675ad0312f8fd43850b581f8/api/listener.go#L43

File: api/listener_server_test.go [hunk 2]
```diff
@@ -103,3 +103,56 @@ func TestListenerServer(t *testing.T) {
...
 }
+
+func TestDeleteActiveListener(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	srv, err := api.NewServer(ctx)
+	require.NoError(t, err)
+
+	rec, err := srv.Upsert(ctx, &pb.Record{
```
Comment (calebdoxsey) [inline api/listener_server_test.go:114]: Could we add a second record to confirm deleting non-connected listener deletion still works?

**Review by calebdoxsey — CHANGES_REQUESTED** [review]

File: api/config_server.go (hunk 1 above)
Comment (wasaga) [reply api/config_server.go:48]: the only error kind that may be returned is that there's no active listener. 
the `disconnectLocked` would fill in `ListenerStatus.LastError` field, which we ignore during record deletion.

File: api/listener_server_test.go (hunk 2 above)
Comment (wasaga) [reply api/listener_server_test.go:114]: added

**Review by calebdoxsey — APPROVED** [review]

**Pull request merged**